frontend/test/pytest/test_static_arguments.py

```diff
@@ -136,6 +137,22 @@ def f(x: int, y: MyClass):
         assert f(1, my_obj) == 9
         assert function != f.compiled_function
 
+    def test_qnode_with_static_arguments(self, capsys):
+        """Test if QJIT static arguments pass through QNode correctly."""
+        dev = qml.device("lightning.qubit", wires=1)
+
+        @qjit(static_argnums=(1,))
+        @qml.qnode(dev)
+        def circuit(x, c):
+            print("Inside QNode:", c)
+            qml.RY(c, 0)
+            qml.RX(x, 0)
+            return qml.expval(qml.PauliZ(0))
+
+        circuit(0.5, 0.5)
+        captured = capsys.readouterr()
+        assert captured.out.strip() == "Inside QNode: 0.5"
+
```

Comment:
I am adding this test case, which I think is also the main motivation behind the ticket.
 
```suggestion
    def test_qnode_nested_with_static_arguments(self, capsys):
        """Test if QJIT static arguments pass through QNode correctly."""
        dev = qml.device("lightning.qubit", wires=1)

        @qjit(static_argnums=(1,))
        @qml.qnode(dev)
        def circuit(x, c):
            print("Inside QNode:", c)
            qml.RY(c, 0)
            qml.RX(x, 0)
            return qml.expval(qml.PauliZ(0))

        @qjit(static_argnums=(1,))
        def wrapper(x, c):
            return circuit(x, c)

        wrapper(0.5, 0.5)
        captured = capsys.readouterr()
        assert captured.out.strip() == "Inside QNode: 0.5"
```

Reply:
Thanks for the suggestion. I added this test as well.